### PR TITLE
[#233] Enable `isMinifyEnabled` for `data`

### PR DESCRIPTION
**Pull request by Tuubz**
#233 

## What happened 👀

Enable `isMinifyEnabled` for `data` in CoroutineTemplate (for release variant)

## Proof Of Work 📹

App is buildable in release variant (after manually adding `release.keystore`):

https://user-images.githubusercontent.com/18277915/173030844-3219dbc3-06ce-4446-ab6f-77acf21d9072.mov

### CODE REVIEW

File: CoroutineTemplate/data/build.gradle.kts [hunk 1]
```diff
@@ -19,11 +19,14 @@ android {
 
     buildTypes {
         getByName(BuildType.RELEASE) {
-            isMinifyEnabled = false
-            proguardFiles(getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro")
+            isMinifyEnabled = true
+            proguardFiles(
+                getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
+            )
```
Comment (luongvo) [inline CoroutineTemplate/data/build.gradle.kts:25]: @Tuubz please also take care of adding a new ticket to handle proguard setup, such as:
- keep Android stuff
- keep retrofit, moshi, okhttp
- keep API data class, enum, Parcelable
- keep logs, etc

**Review by luongvo — APPROVED** [review]

**Review by sleepylee — APPROVED** [review]

File: CoroutineTemplate/data/build.gradle.kts (hunk 1 above)
Comment (Tuubz) [reply CoroutineTemplate/data/build.gradle.kts:25]: @luongvo Hmmmm, do we really need it though? In the screen-recording above, I built a release variant. The only update which was required to run successfully, was to add `@Keep` to `SecondBundle.kt`. So I don't think it's necessary to make an update for sample code.

Comment (Tuubz) [reply CoroutineTemplate/data/build.gradle.kts:25]: I'll test it out again asap, I think you might be right 👍

Comment (Tuubz) [reply CoroutineTemplate/data/build.gradle.kts:25]: Hmmmm, I tried accessing the Jetpack Compose screen from the app (it calls the API)

It's still working fine for me, not sure if I missed something. Would you mind creating an RFC for this then? 🙇

**Review by AVI5HEK — APPROVED** [review]

**Pull request merged**